### PR TITLE
NAS-116083 / 22.12 / Allow using keychain SSH credentials for rsync tasks

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 18:30

**Comment by yocalebo** [comment]
retest this please

**Comment by yocalebo** [comment]
retest this please

File: src/middlewared/middlewared/plugins/rsync.py [hunk 1]
```diff
@@ -642,55 +666,94 @@ async def do_delete(self, id):
...
+
+                    remote = f'"{credentials["username"]}"@{credentials["host"]}'
+                    port = credentials['port']
+
+                    user = self.middleware.call_sync('dscache.get_uncached_user', rsync['user'])
+
+                    private_key_file = exit_stack.enter_context(tempfile.NamedTemporaryFile('w'))
+                    os.chmod(private_key_file.name, 0o600)
+                    os.chown(private_key_file.name, user['pw_uid'], user['pw_gid'])
+                    private_key_file.write(key_pair['attributes']['private_key'])
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/rsync.py:723]: can't we use `fchmod` and `fchown` since we already have a file open?

**Comment by themylogin** [comment]
@anodos325 fixed

**Comment by themylogin** [comment]
@anodos325 ping

**Review by anodos325 — APPROVED** [review]

**Pull request merged**